blockchain/chain_sync/src/network_context.rs

```diff
@@ -70,6 +70,26 @@ impl SyncNetworkContext {
             .map(|fts| fts.tipset().map_err(|e| e.to_string()))
             .collect::<Result<_, _>>()
     }
+    /// Send a blocksync request for full tipsets (includes messages)
+    pub async fn blocksync_fts(
+        &mut self,
+        peer_id: PeerId,
+        tsk: &TipSetKeys,
+    ) -> Result<FullTipset, String> {
+        let bs_res = self
+            .blocksync_request(
+                peer_id,
+                BlockSyncRequest {
+                    start: tsk.cids().to_vec(),
+                    request_len: 1,
+                    options: 3,
+                },
+            )
+            .await?;
+
+        let fts = bs_res.into_result()?;
+        Ok(fts[0].clone())
```

Comment:
```suggestion
        Ok(fts[0])
```

Reply:
Need to clone, the compiler complains otherwise.

Reply:
you're right, for some reason thought it could be moved. One point about this, can you change it to a checked indexing (`get(0)`) or make sure the length is 1 before this so that a bad response won't crash the node?